src/validator/task.rs

```diff
@@ -659,7 +659,7 @@ fn handle_m8(
     transaction: &Transaction,
     accepted_bmm_requests: &BmmCommitments,
     prev_mainchain_block_hash: &BlockHash,
-) -> Result<(), HandleM8Error> {
+) -> Result<Option<()>, HandleM8Error> {
```

Comment:
Why return `Option<()>` when it is isomorphic to `bool`?
`bool` is a better choice here.
Please add a doc comment explaining what the `bool` does, so that the caller knows how to handle it